util/Pending.h

```diff
@@ -127,6 +127,52 @@ namespace Pending {
         return Pending<decltype(parser(path))>(
             std::async(std::launch::async, parser, path), path.filename().string());
     }
+
+    /** Return a Pending<T> constructed with \p parser, \p arg1, and \p path*/
+    template <typename Func, typename Arg1>
+    auto Parsed(const Func& parser, const Arg1& arg1, const boost::filesystem::path& path)
+        -> Pending<decltype(parser(arg1, path))>
+    {
+        auto result = parser(arg1, path);
+        auto promise = std::promise<decltype(parser(arg1, path))>();
+        promise.set_value(std::move(result));
+        return Pending<decltype(parser(arg1, path))>(promise.get_future(), path.filename().string());
+    }
+
+    template <typename Func>
+    struct Parsing {
+        const Func& parser;
```

Comment:
`util\Pending.h(144,1): warning C4180: qualifier applied to function type has no meaning; ignored (compiling source file ..\..\util\AppInterface.cpp)`

Reply:
It looks strange because `StartParsing` above accepts function with qualifiers.